File: java/src/main/java/com/adaptionsoft/games/uglytrivia/domain/Players.java
package com.adaptionsoft.games.uglytrivia.domain;

import com.adaptionsoft.games.uglytrivia.domain.events.PlayerAdded;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collection;
import java.util.List;

public class Players {
    private List<Player> players = new ArrayList<>();
    private int currentPlayer;

    public List<Object> add(String playerName) {
        players.add(new Player(playerName));
        return Arrays.asList(
            new PlayerAdded(playerName, players.size()));
    }

    public Player getCurrentPlayer() {
        return players.get(currentPlayer);
    }

    public List<Object> switchToNextPlayer() {
        List<Object> events = getCurrentPlayer().hasPlayerWon();
        currentPlayer++;
        if (currentPlayer == players.size()) currentPlayer = 0;
        return events;
    }
}

File: php/src/Domain/Players.php
<?php

namespace Trivia\Domain;

class Players
{
    /**
     * @var Player[]
     */
    private $players = array();
    private $currentPlayer = 0;

    /**
     * Players constructor.
     */
    public function __construct()
    {
    }

    public function addPlayer($playerName)
    {
        array_push($this->players, new Player($playerName));
    }

    public function getCurrentPlayer(): Player
    {
        return $this->players[$this->currentPlayer];
    }

    /**
     * @return bool
     */
    public function switchToNextPlayer(): bool
    {
        $winner = $this->players[$this->currentPlayer]->hasPlayerWon();
        $this->currentPlayer++;
        if ($this->currentPlayer == count($this->players)) $this->currentPlayer = 0;

        return $winner;
    }

    public function getPlayersNumber(): int
    {
        return count($this->players);
    }

}